contracts/oracles/lido/LidoOracle.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import '@yield-protocol/utils-v2/contracts/access/AccessControl.sol';
+import '@yield-protocol/utils-v2/contracts/cast/CastBytes32Bytes6.sol';
+import '@yield-protocol/utils-v2/contracts/token/IERC20Metadata.sol';
+import '@yield-protocol/vault-interfaces/IOracle.sol';
+import '../../constants/Constants.sol';
+import './IWstETH.sol';
+
+/**
+ * @title LidoOracle
+ * @notice
+ */
+contract LidoOracle is IOracle, AccessControl, Constants {
+    using CastBytes32Bytes6 for bytes32;
+    IWstETH wstETH;
+    bytes32 public immutable wstEthId;
+    bytes32 public immutable stEthId;
+
+    constructor(bytes32 wstEthId_, bytes32 stEthId_) public {
+        wstEthId = wstEthId_;
+        stEthId = stEthId_;
+    }
+
+    function setSource(IWstETH wstETH_) external auth {
+        wstETH = wstETH_;
+    }
+
+    /**
+     * @notice Retrieve the value of the amount at the latest oracle price.
+     */
+    function peek(
+        bytes32 base,
+        bytes32 quote,
+        uint256 baseAmount
+    ) external view virtual override returns (uint256 quoteAmount, uint256 updateTime) {
+        return _peek(base.b6(), quote.b6(), baseAmount);
+    }
+
+    /**
+     * @notice Retrieve the value of the amount at the latest oracle price. Same as `peek` for this oracle.
+     */
+    function get(
+        bytes32 base,
+        bytes32 quote,
+        uint256 baseAmount
+    ) external virtual override returns (uint256 quoteAmount, uint256 updateTime) {
+        return _peek(base.b6(), quote.b6(), baseAmount);
+    }
+
+    function _peek(
+        bytes6 base,
+        bytes6 quote,
+        uint256 baseAmount
+    ) private view returns (uint256 quoteAmount, uint256 updateTime) {
+        require((base == wstEthId && quote == stEthId) || (base == stEthId && quote == wstEthId), 'Source not found');
+
+        if (base == wstEthId) {
+            //Base equals WstETH
+            quoteAmount = wstETH.getStETHByWstETH(baseAmount);
```

Comment:
Is there any need to handle the possibility of getting back some weird or erroneus data?

`X5 - Is the result checked and errors dealt with? (SWC-104)`

Reply:
How would you recognize erroneous data?

Reply:
@alcueca 0 I guess would prob be erroneous?

Reply:
Not if you pass 0 as a parameter.

I don't think that you can verify the validity of the data received as a general rule. It seems to me a case-by-case basis, where you know that a function could return wrong data instead of reverting. You can see examples of this defensive programming pattern in TransferHelper, for example.

I think that this kind of defense should only be put up when we can't know in advance who will be the contract that we are calling. I.e., when we let users determine the called contract address by letting them use a factory or something.